promql/testdata/functions.test

```diff
@@ -230,11 +230,11 @@ eval instant at 50m predict_linear(testcounter_reset_middle[100m] @ 3000, 3600)
 
 # intercept at t = 600+3600 = 4200
 eval instant at 10m predict_linear(testcounter_reset_middle[100m] @ 3000, 3600)
-	{} 51.36363636363637
+	{} 76.81818181818181
```

Comment:
Note to reviewers: the result of evaluating `predict_linear` as an instant query at 50m (== 3000s) should be what is returned for other evaluation timestamps with the `@ 3000` modifier.

Reply:
I think that's not what we want to see. From the discussion in the issue, I conclude that we want to use the slope as calculated `@ 3000`, i.e. the same slope as for the two examples above, but extrapolate to a point in time that is eval time (10m) plus 3600s, i.e. t=600+3600=4200, which is 51.36363636363637 - interestingly what the test was yielding before your change.

I think the test is just not running into the problem that is described in #11708 because the testing framework is not really testing range queries.

To be precise, what it does: It first runs the instant query. But instant queries never have the problem in #11708. Then it runs a range query with a 1m step and three evaluations, one at 1m before the original eval time, one at the eval time, and one 1m after the eval time. Then it checks that the middle step is still yielding the same result.

In this case, the range selector is long enough to always contain enough points because we only see a time shift of 1m. However, you can trick the test by picking a range of exactly 50m (instead of 100m). Then the first test (line 224) still succeeds, but the 2nd test (line 228) fails because the 2nd step of the range evaluation (which is used for the test) is missing the first value now. If you read the test output carefully, you can even see that it only fails in "range mode", so the instant query still succeeds.

Still, your changes might be on the right track, just that you need to re-include the extrapolation.

I propose the following:
- Change the tests back to the originally expected values (they were right), but change the range from `[100m]` to `[50m]`
- Then make the tests pass.